File: Day 28 : RegEx, Patterns, and Intro to Databases/Solution.java
import java.io.*;
import java.util.*;
import java.util.regex.*;

public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(bufferedReader.readLine());
        ArrayList<String> arr = new ArrayList<String>();
        String regString = "[a-zA-z0-9]@gmail.com";
        Pattern p = Pattern.compile(regString);

        for(int i = 0; i < N; i++){
            try {
                String[] firstMultipleInput = bufferedReader.readLine().replaceAll("\\s+$", "").split(" ");

                String firstName = firstMultipleInput[0];

                String emailID = firstMultipleInput[1];

                Matcher m = p.matcher(emailID);
                if (m.find()) {
                    arr.add(firstName);
                }
            } catch (IOException ex) {
                throw new RuntimeException(ex);
            }
        };

        Collections.sort(arr);

        for (String name : arr) {
            System.out.println(name);
        }
        
        bufferedReader.close();
    }
}

File: Day 28 : RegEx, Patterns, and Intro to Databases/Solution.js
function main() {
    const N = parseInt(readLine().trim(), 10);
    
    let nameArr = [];
    
    for (let NItr = 0; NItr < N; NItr++) {
        const firstMultipleInput = readLine().split(' ');

        const firstName = firstMultipleInput[0];

        const emailID = firstMultipleInput[1];
        
        let pattern = /[\s]*@gmail.com/g;
        
        if(pattern.test(emailID)){
            nameArr.push(firstName);
        }
    }
    
    nameArr.sort();
    
    for(let i = 0; i < nameArr.length; i++) {
        console.log(nameArr[i]);
    }
}

main();
